core/controller/src/main/scala/org/apache/openwhisk/core/controller/Controller.scala

```diff
@@ -176,6 +184,43 @@ class Controller(val instance: ControllerInstanceId,
     LogLimit.config,
     runtimes,
     List(apiV1.basepath()))
+
+  private val controllerCredentials = loadConfigOrThrow[ControllerCredentials](ConfigKeys.controllerCredentials)
+
+  /**
+   * config user memory of ContainerPool
+   */
+  import org.apache.openwhisk.core.connector.ConfigMemoryProtocol._
+  private val configMemory = {
+    implicit val executionContext = actorSystem.dispatcher
+    (path("config" / "memory") & post) {
+      extractCredentials {
+        case Some(BasicHttpCredentials(username, password)) =>
+          if (username == controllerCredentials.username && password == controllerCredentials.password) {
+            entity(as[String]) { memory =>
+              val configMemoryList = memory.parseJson.convertTo[ConfigMemoryList]
+
+              val existIllegalUserMemory = configMemoryList.items.exists { configMemory =>
```

Comment:
Scalaism. I think we can change this something like this:

configMemoryList.items.find(MemoryLimit.MIN_MEMORY.compare(_.memory) > 0)) match {
 case Some(_) => 
   // reject the request
 case None =>
}